src/init/features/frameworks/repo.ts

```diff
@@ -54,70 +78,128 @@ export async function linkGitHubRepository(
+  if (existingConns.length < 1) {
+    let oauthConn = await getOrCreateConnection(projectId, location, FRAMEWORKS_OAUTH_CONN_NAME);
+    while (oauthConn.installationState.stage === "PENDING_USER_OAUTH") {
+      oauthConn = await promptConnectionAuth(oauthConn);
+    }
+    // Create or get connection resource that contains reference to the GitHub oauth token.
+    // Oauth token associated with this connection should be used to create other connection resources.
+    const connectionId = generateConnectionId();
+    const conn = await createConnection(projectId, location, connectionId, {
+      authorizerCredential: oauthConn.githubConfig?.authorizerCredential,
+    });
+    let refreshedConn = conn;
+    while (refreshedConn.installationState.stage !== "COMPLETE") {
+      refreshedConn = await promptAppInstall(conn);
+    }
+    existingConns.push(refreshedConn);
+  }
 
-  let remoteUri = await promptRepositoryURI(projectId, location, connectionId);
+  let { remoteUri, connection } = await promptRepositoryUri(projectId, location, existingConns);
   while (remoteUri === "") {
     await utils.openInBrowser("https://github.com/apps/google-cloud-build/installations/new");
     await promptOnce({
       type: "input",
       message:
         "Press ENTER once you have finished configuring your installation's access settings.",
     });
-    remoteUri = await promptRepositoryURI(projectId, location, connectionId);
+    const selection = await promptRepositoryUri(projectId, location, existingConns);
+    remoteUri = selection.remoteUri;
+    connection = selection.connection;
   }
 
+  // Ensure that the selected connection exists in the same region as the backend
+  const { id: connectionId } = parseConnectionName(connection.name)!;
+  await getOrCreateConnection(projectId, location, connectionId, {
+    authorizerCredential: connection.githubConfig?.authorizerCredential,
+    appInstallationId: connection.githubConfig?.appInstallationId,
+  });
   const repo = await getOrCreateRepository(projectId, location, connectionId, remoteUri);
   logger.info();
   utils.logSuccess(`Successfully linked GitHub repository at remote URI:\n ${remoteUri}`);
   return repo;
 }
 
-async function promptRepositoryURI(
+async function promptRepositoryUri(
   projectId: string,
   location: string,
-  connectionId: string
-): Promise<string> {
-  const resp = await gcb.fetchLinkableRepositories(projectId, location, connectionId);
-  if (!resp.repositories || resp.repositories.length === 0) {
-    throw new FirebaseError(
-      "The GitHub App does not have access to any repositories. Please configure " +
-        "your app installation permissions at https://github.com/settings/installations."
-    );
+  connections: gcb.Connection[]
+): Promise<{ remoteUri: string; connection: gcb.Connection }> {
+  const remoteUriToConnection: Record<string, gcb.Connection> = {};
+  for (const conn of connections) {
```

Comment:
Shouldn't all connections have the same oauth credential and thus have access to the same repositories (I actually can't remember if this is true or if connections use a different resource to fetch linkable repositories)? Or is it possible that we have connections containing different oauth tokens?

Reply:
Yes to same auth credentials, but we may have connections that point to different installation - e.g. repos in Firebase org account AND repos in personal account.